docs: i18n interpolation

smhutch opened the pull request:
### Summary

Added minimal documentation for value interpolation via `Eui18n` and `useEuiI18n`. An additional example of component interpolation could also be added here if required (as requested in #4044). I was unsure if it would be preferred to add a new section + examples, or combine that example into my existing one. This additional example would highlight that passing components as values (to interpolate a `EuiButton`, or `strong` tag, for example) works. Please let me know what's preferred.

### Checklist

- [ ] ~Check against **all themes** for compatibility in both light and dark modes~
- [ ] ~Checked in **mobile**~
- [ ] ~Checked in **Chrome**, **Safari**, **Edge**, and **Firefox**~
- [ ] ~Props have proper **autodocs**~
- [x] Added **[documentation](https://github.com/elastic/eui/blob/master/wiki/documentation-guidelines.md)**
- [x] Checked **[Code Sandbox](https://codesandbox.io/)** works for the any docs examples
- [ ] ~Added or updated **[jest tests](https://github.com/elastic/eui/blob/master/wiki/testing.md)**~
- [ ] ~Checked for **breaking changes** and labeled appropriately~
- [ ] ~Checked for **accessibility** including keyboard-only and screenreader modes~
- [ ] ~A **[changelog](https://github.com/elastic/eui/blob/master/wiki/documentation-guidelines.md#changelog)** entry exists and is marked appropriately~

Closes #3835
Related to #4044

kibanamachine commented:
Since this is a community submitted pull request, a Jenkins build has not been kicked off automatically. Can an Elastic organization member please verify the contents of this patch and then kick off a build manually?

chandlerprall commented:
Thanks for this! I'll run CI to get a docs deployment and check it out. jenkins test this

kibanamachine commented:
Preview documentation changes for this PR: https://eui.elastic.co/pr_4100/

chandlerprall commented:
Adding this as a new section makes sense - couple of thoughts:

* would be great to include a version injecting a component as well, as you mentioned
* while the Demo JS tab shows how this is used, including an example interpolation in the section's text - _useEuiI18n and EuiI18n support variable interpolation._ would be good to see

smhutch commented:
@chandlerprall, thanks for the guidance. I have rebased to update this PR. Latest changes are:

1. I fixed a few lint-warnings (semi-colons, trailing commas, single quotes)
2. I added two additional examples, both of which show component interpolation. For now I included these within the same code example and section.
3. I expanded the section text to include an additional example of how to interpolate a string. I'm not 100% sure if this is what you had in mind, so please let me know 🙂

chandlerprall commented:
jenkins test this

kibanamachine commented:
Preview documentation changes for this PR: https://eui.elastic.co/pr_4100/

chandlerprall reviewed (approved):
Documentation, snippets, demo code & execution, and code sandbox all look good. Thank you very much @smhutch !
